metals/src/main/scala/scala/meta/internal/metals/OnTypeFormattingProvider.scala

```diff
@@ -0,0 +1,71 @@
+package scala.meta.internal.metals
+
+import org.eclipse.lsp4j.{DocumentOnTypeFormattingParams, Range, TextEdit}
+
+import scala.meta.internal.mtags.Semanticdbs
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.meta.inputs.Input
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.tokens.Token
+import scala.meta.tokens.Token.Constant
+
+final class OnTypeFormattingProvider(
+    semanticdbs: Semanticdbs
+)(implicit ec: ExecutionContext) {
+
+  val quot = """\u0022\u0022\u0022"""
+  val space = " "
+
+  private def indent(toInput: Input.VirtualFile, pos: meta.Position): String = {
+    val bit = toInput.text.substring(0, pos.start)
+    val lastNewline = bit.lastIndexOf("\n")
+    val lastPipe = bit.lastIndexOf("|")
+    val length = toInput.text
+      .substring(lastNewline, lastPipe)
+      .length - 1
+    space * length
+  }
+
+  private def longString(text: String, token: Token): Boolean = {
+    text.substring(token.start, token.start + 3).equals(quot)
+  }
+
+  private def inToken(pos: meta.Position, token: Token): Boolean = {
+    pos.start >= token.start && pos.end <= token.end
+  }
+
+  private def pipeInScope(pos: meta.Position, text: String): Boolean = {
+    val lastNew =
+      text.substring(0, pos.start).lastIndexOf("\n")
+    text
+      .substring(lastNew, pos.start)
+      .contains("|")
+  }
+
+  def format(
+      params: DocumentOnTypeFormattingParams
+  ): Future[Option[TextEdit]] = {
+    val source = params.getTextDocument.getUri.toAbsolutePath
+    if (source.exists) {
+      val range = new Range(params.getPosition, params.getPosition)
+      val pos = params.getPosition.toMeta(source.toInput)
+      val sourceText = source.toInput.text
+      val tokens = source.toInput.tokenize.toOption
+      var edit: Option[TextEdit] = None
+      tokens.get.foreach {
```

Comment:
why `get` ? what if there are no tokens? maybe `getOrElse(Nil)` would be better?
or something like this?
```scala
if(pipeInScope(pos, sourceText){
  val tokenized = source.toInput.tokenize.toOption
  tokenized.flatMap{ tokens => 
    tokens.collectFirst{ 
      case token: Constant.String if inToken(pos, token) && longString(sourceText, token) => 
        new TextEdit(range, indent(source.toInput, pos) + "|")
      }
  }
} else {
  None
}
```